src/lib/state.ts

```diff
@@ -24,46 +42,82 @@ interface CommandClassState {
   isSecure: boolean;
 }
 
-type EndpointState = Partial<Endpoint>;
-
-type DeviceClassState = Partial<
-  Modify<
-    DeviceClass,
-    {
-      basic: {
-        key: number;
-        label: string;
-      };
-      generic: {
-        key: number;
-        label: string;
-      };
-      specific: {
-        key: number;
-        label: string;
-      };
-    }
-  >
->;
+interface EndpointState {
+  nodeId: Endpoint["nodeId"];
+  index: Endpoint["index"];
+  installerIcon: Endpoint["installerIcon"];
+  userIcon: Endpoint["userIcon"];
+}
+
+interface DeviceClassState {
+  basic: {
+    key: number;
+    label: string;
+  };
+  generic: {
+    key: number;
+    label: string;
+  };
+  specific: {
+    key: number;
+    label: string;
+  };
+  mandatorySupportedCCs: DeviceClass["mandatorySupportedCCs"];
+  mandatoryControlledCCs: DeviceClass["mandatoryControlledCCs"];
+}
 
 interface ValueState extends TranslatedValueID {
   metadata: ValueMetadata;
   ccVersion: number;
   value?: any;
 }
 
-type NodeState = Partial<
-  Modify<
-    ZWaveNode,
-    {
-      deviceClass: DeviceClass | DeviceClassState;
-      commandClasses: CommandClassState[];
-      endpoints: EndpointState[];
-      values: ValueState[];
-    }
-  >
+interface NodeState0 {
```

Comment:
I didn't realize at first where the 0 was coming from. For external reviewers or later contributors it may be better to better describe this and name it something like "NodeState_Schema0" but maybe this is just me.